doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+|                    |                 | - Retains all properties.                       |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * A)(x)``     | ``f(A(x))``     | `FunctionalComp`                                |
+|                    |                 | - Retains gradient                              |
++--------------------+-----------------+-------------------------------------------------+
+| ``(a * f)(x)``     | ``a * f(x)``    | `FunctionalLeftScalarMult`                      |
+|                    |                 | - Retains all properties, if ``a`` is positive. |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * a)(x)``     | ``f(a * x)``    | `FunctionalRightScalarMult`                     |
+|                    |                 | - Retains all properties                        |
++--------------------+-----------------+-------------------------------------------------+
+| ``(v * f)(x)``     | ``v * f(x)``    | `FunctionalLeftVectorMult`                      |
+|                    |                 | - This is not a functional, but an operator     |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * v)(x)``     | ``f(v * x)``    | `FunctionalRightVectorMult`                     |
+|                    |                 | - Retains gradient and convex conjugate.        |
++--------------------+-----------------+-------------------------------------------------+
+| ``f.translate(y)`` | ``f(. - y)``    | `TranslatedFunctional`                          |
+|                    |                 | - Retains all properties.                       |
++--------------------+-----------------+-------------------------------------------------+
+
+
+Code example
+============
+This section contains an example of an implementation of a functional, namely the functional :math:`\|x\|_2^2 + \langle x, y \rangle`.
+Another example can be found `functional_basic_example.py`, and more implementations of other functionals can be found in `default_functionals.py`. ::
+
+    # Here we define the functional
+    class MyFunctional(odl.solvers.Functional):
+        """This is my functional: ||x||_2^2 + <x, y>."""
+
+        def __init__(self, domain, y):
+            # This comand calls the init of Functional and sets a number of
+            # parameters associated with a functional. All but domain have default
+            # values if not set.
+            super().__init__(domain=domain, linear=False, convex=True,
+                             concave=False, smooth=True, grad_lipschitz=2)
+
+            # We need to check that y is in the domain. Then we store the value of
+            # y for future use.
+            if y not in domain:
+                raise TypeError('y is not in the domain!')
+            self._y = y
```

Comment:
For the example I'd simply store `self.y = y` and not introduce an additional property.